openfe/protocols/openmm_utils/charge_generation.py

```diff
@@ -408,12 +414,106 @@ def assign_offmol_partial_charges(
         generate_n_conformers=generate_n_conformers,
     )  # type: ignore
 
-    # Call selected method to assign partial charges
-    CHARGE_METHODS[method.lower()]['charge_func'](
-        offmol=offmol_copy,
-        toolkit_registry=toolkits,
-        **CHARGE_METHODS[method.lower()]['charge_extra_kwargs'],
-    )  # type: ignore
+    # limit the number of threads used by SQM
+    # <https://github.com/openforcefield/openff-toolkit/issues/1831>
+    with temp_env({"OMP_NUM_THREADS": "2"}):
```

Comment:
There's some more pythonic tools for thread limiting like this, let me have a dig in the morning.

i.e. we'll need them going forward, so maybe we should bite the bullet and go for it.